lib/mercadopago/collection.rb

```diff
@@ -100,14 +102,41 @@ def self.notification_preapproval(access_token, preapproval_id)
     #       Active subscription recurring payment.
     #     subscription_payment::
     #       Subscription fee.
-    #
-    def self.search(access_token, search_hash = {}, sandbox = false)
+
+    def self.search(access_token, search_hash = {}, sandbox = false, auto_paginate = false)
       query = search_hash.map { |e| e.join('=') }.join('&')
 
       uri_prefix = sandbox ? '/sandbox' : ''
-      MercadoPago::Request.wrap_get("#{uri_prefix}/v1/payments/search?access_token=#{access_token}&#{query}", { accept: 'application/json' })
+
+      result = []
+
+      if auto_paginate
+        response = get_collection(access_token, query, uri_prefix)
+
+        if response
+          total_items = response.fetch("paging", {}).fetch("total", nil)
+          pages       = total_items ? (total_items / RESULTS_PER_PAGE.to_f).ceil : 1
+
+          pages.times do |page|
+            offset = page * RESULTS_PER_PAGE
+            query  = search_hash.map { |e| e.join('=') }.join('&')
+            query += "&offset=#{offset}" if offset > 0
+
+            response = get_collection(access_token, query, uri_prefix) if offset > 0
+
+            result   << response["results"]
+            result   = [result] unless result.is_a? Array
+          end
+        end
+      else
+        result << get_collection(access_token, query, uri_prefix)
+      end
```

Comment:
@cleicar Ideally we would want this to be implemented with an `Enumerable` that can lazily load all the records. 

This implementation solves the problem but it is not efficient. I think it's good enough though. Maybe you can create an issue to make these changes lazy-load in the future?

You can see an example of a lazy-loading implementation over here: https://github.com/ombulabs/harvesting/blob/7297a558295b21742eafe85cd16a57b2a1fd5a2b/lib/harvesting/enumerable.rb

Reply:
Good point @etagwerker. I just created a new issue for that suggestion.